packages/datatrak-web/src/views/AccountSettingsPage/PersonalDetailsSection.tsx

```diff
@@ -4,10 +4,168 @@
+  function onSubmit(userDetails): SubmitHandler<PersonalDetailsFormFields> {
+    const updates = Object.keys(dirtyFields)
+      .filter(field => dirtyFields[field])
+      .reduce(
+        (updatedFields, field) => ({ ...updatedFields, [field]: userDetails[field] }),
+        {} as UserAccountDetails,
+      );
+
+    updateUser(updates);
+  }
+
   return (
-    <AccountSettingsSection title="Personal details" description="Edit your personal details" />
+    <AccountSettingsSection title="Personal details" description="Edit your personal details">
+      <PersonalDetailsForm onSubmit={handleSubmit(onSubmit)}>
+        <Controller
+          control={control}
+          name="firstName"
+          as={
+            <StyledTextField
+              label="First name"
+              placeholder="First name"
+              autoComplete="given-name"
+              inputProps={{ enterKeyHint: 'next' }}
+              {...register('firstName', { required: true })}
+              required
+            />
+          }
+        />
+        <Controller
+          control={control}
+          name="lastName"
+          as={
+            <StyledTextField
+              label="Last name"
+              placeholder="Last name"
+              autoComplete="family-name"
+              inputProps={{ enterKeyHint: 'next' }}
+              {...register('lastName', { required: true })}
+              required
+            />
+          }
+        />
+        <StyledTextField
+          name="email"
+          label="Email"
+          placeholder="Email"
+          tooltip="You cannot change your email address"
+          autoComplete="email"
+          value={user.email}
+          required
+          disabled
+        />
+        <Controller
+          control={control}
+          name="mobileNumber"
+          as={
+            <StyledTextField
+              label="Contact number (optional)"
+              placeholder="Contact number"
+              autoComplete="tel"
+              inputProps={{ enterKeyHint: 'next', inputMode: 'tel' }}
+              {...register('mobileNumber', { required: true })}
+            />
+          }
+        />
+        <Controller
+          control={control}
+          name="employer"
+          as={
+            <StyledTextField
+              label="Employer"
+              placeholder="Employer"
+              autoComplete="organization"
+              inputProps={{ enterKeyHint: 'next' }}
+              {...register('employer', { required: true })}
+              required
+            />
+          }
+        />
+        <Controller
+          control={control}
+          name="position"
+          as={
+            <StyledTextField
+              label="Position"
+              placeholder="Position"
+              autoComplete="organization-title"
+              inputProps={{ enterKeyHint: 'done' }}
+              {...register('position', { required: true })}
+              required
+            />
+          }
+        />
```

Comment:
Thank you for the sample code! I’d tried it earlier but didn’t know to pass `register` as `inputRef`, so nothing worked 😅

All fields are now uncontrolled

Reply:
Hm, seems that my uncontrolled implementation breaks the logic that disables the <kbd>Save changes</kbd> button when the form is not dirty. I’ve pushed to the branch; any idea why this might be?